src/status_im/contexts/wallet/send/events.cljs

```diff
@@ -491,26 +489,48 @@
                                              :error error}))}]}))
 
-(rf/reg-event-fx :wallet/handle-suggested-routes
+(defn- bridge-amount-greater-than-bonder-fees?
+  [{{token-decimals :decimals} :from-token
+    bonder-fees                :tx-bonder-fees
+    amount-in                  :amount-in}]
+  (let [bonder-fees      (utils.money/token->unit bonder-fees token-decimals)
+        amount-to-bridge (utils.money/token->unit amount-in token-decimals)]
+    (> amount-to-bridge bonder-fees)))
+
+(defn- remove-multichain-routes
+  [routes]
+  (if (> (count routes) 1)
+    [] ;; if route is multichain, we remove it
+    routes))
+
+(defn- remove-invalid-bonder-fees-routes
+  [routes]
+  (filter bridge-amount-greater-than-bonder-fees? routes))
+
+(defn- ->old-route-paths
+  [routes]
+  (map data-store/new->old-route-path routes))
+
+(rf/reg-event-fx
+ :wallet/handle-suggested-routes
  (fn [_ data]
-   (if-let [error (some-> data
-                          first
-                          :ErrorResponse
-                          (#(if (= (:code %) "0") "An error occurred" (:details %))))]
-     (do
+   (if-let [{:keys [code details]} (-> data :ErrorResponse first)]
+     (let [error-message (if (= code "0") "An error occurred" details)]
        (log/error "failed to get suggested routes (async)"
                   {:event :wallet/handle-suggested-routes
-                   :error error})
-       {:fx [[:dispatch [:wallet/suggested-routes-error error]]]})
-     (let [suggested-routes-new-data (data-store/rpc->suggested-routes data)
-           suggested-routes          (-> suggested-routes-new-data
-                                         first
-                                         ;; if route is multichain, we remove it
-                                         (update :best (fn [best] (if (> (count best) 1) [] best)))
-                                         (update :best #(map data-store/new->old-route-path %))
-                                         (update :candidates #(map data-store/new->old-route-path %)))]
-       {:fx [[:dispatch
-              [:wallet/suggested-routes-success suggested-routes]]]}))))
+                   :error error-message})
+       {:fx [[:dispatch [:wallet/suggested-routes-error error-message]]]})
+     (let [best-routes-fix (comp ->old-route-paths
+                                 remove-invalid-bonder-fees-routes
+                                 remove-multichain-routes)
+           candidates-fix  (comp ->old-route-paths
+                                 remove-invalid-bonder-fees-routes)
+           routes          (-> data
+                               (first)
+                               (data-store/rpc->suggested-routes)
+                               (update :best best-routes-fix)
+                               (update :candidates candidates-fix))]
+       {:fx [[:dispatch [:wallet/suggested-routes-success routes]]]}))))
```

Comment:
This is the important change.

The `:best` and `:candidates` routes are updated to remove the ones having `:tx-bonder-fees` higher than the amount to bridge.

I tested and the tranasctions worked. @briansztamfater Do you think we need to add an additional check to perform these updates?
As far as I tested everything was still working.

Reply:
I think the checks are okay!

Reply:
Hey @ulisesmac, just curious why are we using `first` here on `data`:
* What's the shape of data at this point, would we only ever care about the first entry?